### PR TITLE
Use codebase in pkgdef

### PR DESCRIPTION
**Pull request by jmarolf**
This should fix command registration

### CODE REVIEW

File: src/Templates/VS2017/Roslyn.SDK.VS2017.csproj [hunk 1]
```diff
@@ -11,6 +11,7 @@
     <!-- VS Insertion -->
     <VisualStudioInsertionComponent>Microsoft.CodeAnalysis.SDK</VisualStudioInsertionComponent>
     <Ngen>false</Ngen>
+    <UseCodeBase>true</UseCodeBase>
```
Comment (sharwell) [inline src/Templates/VS2017/Roslyn.SDK.VS2017.csproj]: ❓ Why is this needed with `<GeneratePkgDefFile>false</GeneratePkgDefFile>`?

File: src/Tools/SyntaxVisualizer/Roslyn.SyntaxVisualizer.Extension/Roslyn.SyntaxVisualizer.Extension.csproj [hunk 2]
```diff
@@ -15,6 +15,7 @@
     <ExtensionInstallationFolder>Microsoft\SyntaxVisualizer</ExtensionInstallationFolder>
     <VisualStudioInsertionComponent>Roslyn.SyntaxVisualizer.Extension</VisualStudioInsertionComponent>
     <Ngen>false</Ngen>
+    <UseCodeBase>true</UseCodeBase>
```
Comment (tmat) [inline src/Tools/SyntaxVisualizer/Roslyn.SyntaxVisualizer.Extension/Roslyn.SyntaxVisualizer.Extension.csproj]: Unrelated to this change but ` <IsProductComponent>false</IsProductComponent>` seems wrong since the project is also setting `VisualStudioInsertionComponent`. Arcade will override `IsProductComponent` if `VisualStudioInsertionComponent` is set.

Comment (tmat) [reply src/Tools/SyntaxVisualizer/Roslyn.SyntaxVisualizer.Extension/Roslyn.SyntaxVisualizer.Extension.csproj]: Another unrelated thing:

```
<!-- workaround for https://github.com/NuGet/Home/issues/5894 -->
  <Import Project="$(ArtifactsObjDir)$(AssemblyName)\*.props" Condition="'$(MSBuildProjectExtension)' == '.tmp_proj'" />	  <Import Project="$(ArtifactsObjDir)$(AssemblyName)\*.props" Condition="'$(MSBuildProjectExtension)' == '.tmp_proj'" />
```

```
 <!-- workaround for https://github.com/NuGet/Home/issues/5894 -->
  <Import Project="$(ArtifactsObjDir)$(AssemblyName)\*.targets" Condition="'$(MSBuildProjectExtension)' == '.tmp_proj'" />	  <Import Project="$(ArtifactsObjDir)$(AssemblyName)\*.targets" Condition="'$(MSBuildProjectExtension)' == '.tmp_proj'" />
```

shouldn't be needed and also has no effect on .NET Framework 4.7.2, since the WPF temp proj extension changed there. Arcade SDK implements a workaround for the WPF temp projects, so you should be able to just delete these.

Comment (tmat) [inline src/Tools/SyntaxVisualizer/Roslyn.SyntaxVisualizer.Extension/Roslyn.SyntaxVisualizer.Extension.csproj]: I'm a bit confused here by `CreateVsixContainer` being `false`. This project doesn't build VSIX?

Comment (jmarolf) [reply src/Tools/SyntaxVisualizer/Roslyn.SyntaxVisualizer.Extension/Roslyn.SyntaxVisualizer.Extension.csproj]: No, the Roslyn.SDK.2017 project embeds this dll inside of it.  It used to be a VSIX at some point.  I need to go refactor this to be more clear.

File: src/Templates/VS2017/Roslyn.SDK.VS2017.csproj (hunk 1 above)
Comment (jmarolf) [reply src/Templates/VS2017/Roslyn.SDK.VS2017.csproj]: @tmat 	for the answer...

Comment (jmarolf) [reply src/Templates/VS2017/Roslyn.SDK.VS2017.csproj]: Well I'll take a stab at it.  We don't generate a pkgdef for this project but we do for its dependencies.  I'll do some testing, perhaps this is not required on this project.

File: src/Tools/SyntaxVisualizer/Roslyn.SyntaxVisualizer.Extension/Roslyn.SyntaxVisualizer.Extension.csproj (hunk 2 above)
Comment (tmat) [reply src/Tools/SyntaxVisualizer/Roslyn.SyntaxVisualizer.Extension/Roslyn.SyntaxVisualizer.Extension.csproj]: So, then should `VisualStudioInsertionComponent`, `Ngen`, `ExtensionInstallationFolder`, etc. be set?

Comment (jmarolf) [reply src/Tools/SyntaxVisualizer/Roslyn.SyntaxVisualizer.Extension/Roslyn.SyntaxVisualizer.Extension.csproj]: Unsetting them causes the assemblies to be NGEN'd

Comment (tmat) [reply src/Tools/SyntaxVisualizer/Roslyn.SyntaxVisualizer.Extension/Roslyn.SyntaxVisualizer.Extension.csproj]: Setting `VisualStudioInsertionComponent` does. If this project doesn't produce VSIX then it shouldn't set `VisualStudioInsertionComponent`.
